oss-cloud-backend/handler.js

```diff
@@ -22,22 +24,49 @@ module.exports.hello = async event => {
 // {
 //    username: string  
 // }
-module.exports.getPulls = (event, context, callback) => {
-  var body = JSON.parse(event.body);
-  getPullsHandler.getForkedRepos(body.username) // get all repos of a user
-    .then(repos => getPullsHandler.getRepoDetails(repos))
-    .then(repos => getPullsHandler.getParentRepos(repos))
-    .then(repos => getPullsHandler.getUserPulls(body.username, repos))
-    .then(pulls => callback(null, pulls))
-    .catch(err => {
-      console.log("error in getPulls handler: ", err);
-      let response = {
-        statusCode: err.status,
-        body: JSON.stringify({
-          message: err.message
-        })
-      }
-      callback(null, response);
-    })
+module.exports.getPullRequests = async (event, context, callback) => {
+  let response;
+  // check if request is valid
+  try {
+    var body = JSON.parse(event.body);
+    body.username;
+  } catch {
+    response = {
+      statusCode: 500,
```

Comment:
If the JSON format is bad that's a client side error so the status code should be 4XX

Reply:
Changed response status to 400.